pkg/api/e2e_test.go

```diff
@@ -0,0 +1,93 @@
+//go:build integration
+// +build integration
+
+package api_test
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.anx.io/go-anxcloud/pkg/api"
+	clouddnsv1 "go.anx.io/go-anxcloud/pkg/apis/clouddns/v1"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+	vlanv1 "go.anx.io/go-anxcloud/pkg/apis/vlan/v1"
+	"go.anx.io/go-anxcloud/pkg/client"
+	"go.anx.io/go-anxcloud/pkg/utils/test"
+	testutils "go.anx.io/go-anxcloud/pkg/utils/test"
+)
+
+const (
+	waitTimeout  = 5 * time.Minute
+	retryTimeout = 15 * time.Second
+)
+
+var _ = Describe("options", func() {
+	var a api.API
+
+	BeforeEach(func() {
+		var err error
+		a, err = api.NewAPI(
+			api.WithClientOptions(client.AuthFromEnv(false)),
+		)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("AutoTag", func() {
+		It("can automatically tag resources on api.Create", func() {
+			vlan := vlanv1.VLAN{
+				DescriptionCustomer: "go-anxcloud test api.Create AutoTag " + testutils.RandomHostname(),
+				Locations: []corev1.Location{
+					{Identifier: "52b5f6b2fd3a4a7eaaedf1a7c019e9ea"},
+				},
+			}
+
+			DeferCleanup(func() {
+				if vlan.Identifier == "" {
+					return
+				}
```

Comment:
~we can go without this, as we only clean up in `DeferCleanup`~

ah wait, we need this for when VLAN creation fails